File: implementations/node/sdk/src/contracts/executions/execution-submission-request.ts
import type { AiSdkJsonValue } from "../../json/json-value.js";
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";

export interface AiSdkExecutionSubmissionRequest {
  readonly schemaVersion?: typeof AI_SDK_SCHEMA_VERSIONS.executionSubmissionRequest;
  readonly publicationRef: string;
  readonly idempotencyKey?: string;
  readonly input?: AiSdkJsonValue;
  readonly metadata?: Readonly<Record<string, string>>;
  readonly correlationId?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/executions/execution_submission_request.py
from __future__ import annotations

from dataclasses import dataclass, field

from ...json_types import AiSdkJsonValue
from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions


@dataclass(frozen=True)
class AiSdkExecutionSubmissionRequest(AiSdkWireModel):
    publication_ref: str
    schema_version: int = AiSdkSchemaVersions.EXECUTION_SUBMISSION_REQUEST
    idempotency_key: str | None = None
    input: AiSdkJsonValue | None = field(default=None, metadata={"json": True})
    metadata: dict[str, str] = field(default_factory=dict)
    correlation_id: str | None = None
